plinth/__main__.py

```diff
@@ -285,6 +292,17 @@ def list_dependencies(module_list):
     sys.exit(error_code)
 
 
+def list_modules() :
```

Comment:
We can have list_modules take one argument modules_type, which can be either None, or "essential", or "optional".